arch/posix/CMakeLists.txt

```diff
@@ -30,6 +32,17 @@ zephyr_link_libraries_ifdef(CONFIG_COVERAGE
 	-lgcov
 	)
 
+else()
+
+zephyr_compile_options_ifdef(CONFIG_COVERAGE
+	--coverage
+	-fno-inline
+)
+zephyr_link_libraries_ifdef(CONFIG_COVERAGE
+	--coverage
+)
+endif()
+
```

Comment:
We use a different mechanism to achieve portability.

See

https://github.com/zephyrproject-rtos/zephyr/commit/01592071f1e205e7a020d9dd487c04943226c287

for an example.